db/record.c

```diff
@@ -265,25 +264,30 @@ int add_record(struct ireq *iq, void *trans, const uint8_t *p_buf_tag_name,
 
     if (!(flags & RECFLAGS_NEW_SCHEMA)) { // dont lock if adding from SC
 
-        int d_ms = BDB_ATTR_GET(thedb->bdb_attr, DELAY_LOCK_TABLE_RECORD_C);
+        int d_ms = BDB_ATTR_GET(thedb->bdb_attr, DELAY_WRITES_IN_RECORD_C);
         if (d_ms) {
             if (iq->debug)
-                reqprintf(iq, "Sleeping for %d ms", d_ms);
-            usleep(1000 * d_ms);
+                reqprintf(iq, "Sleeping for DELAY_WRITES_IN_RECORD_C (%dms)", d_ms);
+            int lrc = usleep(1000 * d_ms);
+            if (lrc)
+                reqprintf(iq, "usleep error rc %d errno %d\n", rc, errno);
         }
 
-        rc = bdb_lock_table_read(iq->usedb->handle, trans);
-        if (rc == BDBERR_DEADLOCK) {
-            if (iq->debug)
-                reqprintf(iq, "LOCK TABLE READ DEADLOCK");
-            retrc = RC_INTERNAL_RETRY;
-            ERR;
-        } else if (rc) {
-            if (iq->debug)
-                reqprintf(iq, "LOCK TABLE READ ERROR: %d", rc);
-            *opfailcode = OP_FAILED_INTERNAL;
-            retrc = ERR_INTERNAL;
-            ERR;
+        if(!iq->is_sorese) {
```

Comment:
On second thought, this may not be safe for cascading delete/update? Can you double check?

Reply:
add_record is fine i guess.. I meant for upd_record and del_record

Reply:
Will add flag to make sure we call these functions from a place we know we have gotten a read table lock.